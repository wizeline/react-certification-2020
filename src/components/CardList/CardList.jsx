import { Flex, Grid } from '@chakra-ui/layout'
import { useRouter } from 'next/router'
import React, { useEffect, useState } from 'react'
import fetchData from '../../utils/useFetchData'
import Card from '../Card/Card'

const CardList = ({list, IdVid}) => {
    const [data, setData] = useState()
    const { query } = useRouter()
    const { search } = query
    let result = {}
    useEffect(async () => {
        if (IdVid) {
            result = search ?  await fetchData.get("/search", {params: {q: search}}) : await fetchData.get("/search", {params: {relatedToVideoId: IdVid}})
        } else {
            result = search ?  await fetchData.get("/search", {params: {q: search}}) : await fetchData.get("/search")
        }
        console.log(result.data)
        setData(result.data)

        return function cleanup() {
            search = ""
        }
    },[search])

    return (
        <Grid templateColumns={!list?{"md": "repeat(3, 1fr)", "sm": "repeat(1, 1fr)"}:"repeat(1, 1fr)"} mb={5} gap={1}>
            {!data ? "hola" : data.items.map((item) => {
                return (
                    <Card key={item.id.videoId} item={item}/>         
                )
            })}
        </Grid>
    )
}

export default CardList
