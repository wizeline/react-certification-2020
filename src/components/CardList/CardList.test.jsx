import "@testing-library/jest-dom/extend-expect"
import { render, screen, cleanup, waitFor } from "@testing-library/react"
import fetchData from "../../utils/useFetchData"
import CardList from "./CardList"
const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('Testing CardList', () => {
    afterEach(cleanup)
    it('Testing data result from API', async () => {
        const result = await fetchData.get("/search")
        expect(Object.keys(result).length).not.toBe(0);
    });
    
    it('Testing data result from API with params', async () => {
        const result = await fetchData.get("/search", {params: {q: "hola"}})
        expect(Object.keys(result).length).not.toBe(0)
    });
    
    it('CardList Component', async () => {

        useRouter.mockImplementationOnce(() => ({
            query: { search: '' },
          }))

        
        const { container } = render(<CardList/>)
        expect(container).toMatchSnapshot()

    });
    
    
});
