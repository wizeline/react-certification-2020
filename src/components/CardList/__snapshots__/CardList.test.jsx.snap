// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CardList CardList Component 1`] = `
<div>
  <div
    class="css-1r9y2ds"
  >
    hola
  </div>
</div>
`;
